umh-core/pkg/communicator/actions/providers/metrics_provider_mock.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by MockGen. DO NOT EDIT.
+// Source: github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/communicator/actions/providers (interfaces: MetricsProvider)
+//
+// Generated by this command:
+//
+//	mockgen -destination=metrics_provider_mock.go -package=providers github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/communicator/actions/providers MetricsProvider
+//
+
+// Package providers is a generated GoMock package.
+package providers
+
+import (
+	reflect "reflect"
+
+	fsm "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/fsm"
+	models "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/models"
+	gomock "go.uber.org/mock/gomock"
+)
+
+// MockMetricsProvider is a mock of MetricsProvider interface.
+type MockMetricsProvider struct {
+	ctrl     *gomock.Controller
+	recorder *MockMetricsProviderMockRecorder
+	isgomock struct{}
+}
+
+// MockMetricsProviderMockRecorder is the mock recorder for MockMetricsProvider.
+type MockMetricsProviderMockRecorder struct {
+	mock *MockMetricsProvider
+}
+
+// NewMockMetricsProvider creates a new mock instance.
+func NewMockMetricsProvider(ctrl *gomock.Controller) *MockMetricsProvider {
+	mock := &MockMetricsProvider{ctrl: ctrl}
+	mock.recorder = &MockMetricsProviderMockRecorder{mock}
+	return mock
+}
+
+// EXPECT returns an object that allows the caller to indicate expected use.
+func (m *MockMetricsProvider) EXPECT() *MockMetricsProviderMockRecorder {
+	return m.recorder
+}
+
+// GetMetrics mocks base method.
+func (m *MockMetricsProvider) GetMetrics(payload models.GetMetricsRequest, snapshot fsm.SystemSnapshot) (models.GetMetricsResponse, error) {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "GetMetrics", payload, snapshot)
+	ret0, _ := ret[0].(models.GetMetricsResponse)
+	ret1, _ := ret[1].(error)
+	return ret0, ret1
+}
+
+// GetMetrics indicates an expected call of GetMetrics.
+func (mr *MockMetricsProviderMockRecorder) GetMetrics(payload, snapshot any) *gomock.Call {
```

Comment:
Can we rename this to something else, otherwise both very different calls will be called GetMetrics